src/messages/MOSDForceRecovery.h

```diff
@@ -69,12 +69,36 @@ struct MOSDForceRecovery : public Message {
 
   void encode_payload(uint64_t features) {
     using ceph::encode;
+    if (!HAVE_FEATURE(features, SERVER_MIMIC)) {
+      vector<pg_t> pgs;
+      for (auto pgid : forced_pgs) {
+	pgs.push_back(pgid.pgid);
+      }
+      encode(fsid, payload);
+      encode(pgs, payload);
+      encode(options, payload);
+      return;
+    }
+    header.version = HEAD_VERSION;
+    header.compat_version = COMPAT_VERSION;
```

Comment:
You're setting the HEAD/COMPAT_VERSION to their default values. I think you meant to set them to an old value if we don't have SERVER_MIMIC?

Reply:
yep fixed